pkg/txhandler/simple/simple_transaction_handler.go

```diff
@@ -0,0 +1,473 @@
+	res := sth.policyEngineAPIRequest(ctx, &policyEngineAPIRequest{
+		requestType: policyEngineAPIRequestTypeDelete,
+		txID:        txID,
+	})
+	return res.tx, nil
+}
+
+func (sth *simpleTransactionHandler) GetTransactionByID(ctx context.Context, txID string) (transaction *apitypes.ManagedTX, err error) {
+	tx, err := sth.toolkit.Persistence.GetTransactionByID(ctx, txID)
+	if err != nil {
+		return nil, err
+	}
+	if tx == nil {
+		return nil, i18n.NewError(ctx, tmmsgs.MsgTransactionNotFound, txID)
+	}
+	return tx, nil
+}
+func (sth *simpleTransactionHandler) GetTransactions(ctx context.Context, afterStr, signer string, pending bool, limit int, direction toolkit.SortDirection) (transactions []*apitypes.ManagedTX, err error) {
+	var afterTx *apitypes.ManagedTX
+	if afterStr != "" {
+		// Get the transaction, as we need this to exist to pick the right field depending on the index that's been chosen
+		afterTx, err = sth.toolkit.Persistence.GetTransactionByID(ctx, afterStr)
+		if err != nil {
+			return nil, err
+		}
+		if afterTx == nil {
+			return nil, i18n.NewError(ctx, tmmsgs.MsgPaginationErrTxNotFound, afterStr)
+		}
+	}
+	switch {
+	case signer != "" && pending:
+		return nil, i18n.NewError(ctx, tmmsgs.MsgTXConflictSignerPending)
+	case signer != "":
+		var afterNonce *fftypes.FFBigInt
+		if afterTx != nil {
+			afterNonce = afterTx.Nonce
+		}
+		return sth.toolkit.Persistence.ListTransactionsByNonce(ctx, signer, afterNonce, limit, direction)
+	case pending:
+		var afterSequence *fftypes.UUID
+		if afterTx != nil {
+			afterSequence = afterTx.SequenceID
+		}
+		return sth.toolkit.Persistence.ListTransactionsPending(ctx, afterSequence, limit, direction)
+	default:
+		return sth.toolkit.Persistence.ListTransactionsByCreateTime(ctx, afterTx, limit, direction)
+	}
+}
+
+func (sth *simpleTransactionHandler) createManagedTx(ctx context.Context, txID string, txHeaders *ffcapi.TransactionHeaders, gas *fftypes.FFBigInt, transactionData string) (*apitypes.ManagedTX, error) {
+
+	// The request ID is the primary ID, and should be supplied by the user for idempotence
+	if txID == "" {
+		txID = fftypes.NewUUID().String()
+	}
+
+	// First job is to assign the next nonce to this request.
+	// We block any further sends on this nonce until we've got this one successfully into the node, or
+	// fail deterministically in a way that allows us to return it.
+	lockedNonce, err := sth.assignAndLockNonce(ctx, txID, txHeaders.From)
+	if err != nil {
+		return nil, err
+	}
+	// We will call markSpent() once we reach the point the nonce has been used
+	defer lockedNonce.complete(ctx)
+
+	// Sequencing ID is always generated by us - so we have a deterministic order of transactions
+	// Note: We must allocate this within the nonce lock, to ensure that the nonce sequence and the
+	//       global transaction sequence line up.
+	seqID := apitypes.NewULID()
```

Comment:
I still think this should be in FFTM common code.
Maybe the simplest thing would be to simply provide a `sth.toolkit.NewManagedTX()` constructor.

I know we've discussed making the `ManagedTX` an `interface` rather than a `struct` such that the read-only fields can be separated from the writeable fields. I understand if that's too much for this PR, but still I think

Reply:
Looks like you've got a half finished comment here. But combined with other comments around this area. I can see your viewpoint better. I'll summarise my understanding below.

Reply:
@peterbroadhurst For this specific field `sequence id`, maybe it should be generated in the persistence layer as the value of it is database specific?

Reply:
@peterbroadhurst https://github.com/kaleido-io/firefly-transaction-manager/commit/62c6bcf1b80769d5bdeadb7a2fe0740545b2d027 do you feel this can be a solution?